src/charm.py

```diff
@@ -48,12 +42,13 @@ def sha256(hashable) -> str:
 
 
 class MimirK8SOperatorCharm(CharmBase):
-    """Charm the service."""
+    """A Juju Charmed Operator for Mimir."""
 
     _name = "mimir"
     _http_listen_port = 9009
     _instance_addr = "127.0.0.1"
     _stored = StoredState()
+    _restart = False
```

Comment:
I'm curios why this is a class property:
- Afaik, the value would not persist across charm re-inits (stored state would though). Seems like this should be an instance property?
- Would the following be suitable here?

```python
# list and not generator to avoid short-circuiting
restart = any([self._set_pebble_layer(), self._set_mimir_config(), self._set_alerts()])
```

Reply:
Yes, we could move `_restart` to an instance property.

I generally like the `any()` approach. In this case I did not use it because I found it to be less legible....

```python
        try:
            restart = any([self._set_pebble_layer(), self._set_mimir_config(), self._set_alerts()])
        except BlockedStatusError as e:
            self.unit.status = BlockedStatus(e.message)
            return

        if restart:
            self._container.restart(self._name)
            logger.info("Mimir (re)started")
```


```python
        try:
            self._set_alerts()
            self._set_mimir_config()
            self._set_pebble_layer()
        except BlockedStatusError as e:
            self.unit.status = BlockedStatus(e.message)
            return

        if self._restart:
            self._container.restart(self._name)
            logger.info("Mimir (re)started")
```

Anyway the difference is too subtle. We can use the `any()` approach